spec/concurrent/timer_task_spec.rb

```diff
@@ -243,7 +243,7 @@ def trigger_observable(observable)
       end
 
       it 'notifies all observers on timeout' do
-        subject = TimerTask.new(execution: 0.1, timeout: 0.1) { sleep }
+        subject = TimerTask.new(execution: 500, timeout: 0.1, run_now: true) { sleep }
```

Comment:
Thanks for that extremely detailed description. That makes sense.

Here's a failing test that better exposes the bug I'm seeing with `timeout_interval` not being used.

```ruby
it 'should not timeout' do
  subject = TimerTask.new(execution: 0.1, timeout: 1) { sleep(0.5); 42 }
  subject.add_observer(observer)
  subject.execute
  observer.latch.wait(2)
  expect(observer.ex).to be_nil
  expect(observer.value).to eq(42)
  subject.kill
end

# Failures:
# 
#   1) Concurrent::TimerTask observation should not timeout
#      Failure/Error: expect(observer.ex).to be_nil
#        expected: nil
#             got: #<Concurrent::TimeoutError: Concurrent::TimeoutError>
#      # ./spec/concurrent/timer_task_spec.rb:250:in `block (3 levels) in <module:Concurrent>'
```

Execute a task every 0.1s that takes 0.5s to complete with a 1s timeout. This task will be timed out after 0.1s because `execution_interval` is used to trigger the timeout.

My modification of the existing timeout test was to expose the other side of this same bug. Create a long-running task that runs immediatley with a short `timeout_interval` and long enough `execution_interval` to make it clear it won't be run more than once during the test. I expect the task to time out after `timeout_interval` is exceeded but the result is the test will fail after waiting for the task to complete.